plugin/src/main/java/org/wildfly/plugin/provision/ExecUtil.java

```diff
@@ -0,0 +1,161 @@
+// copied and adapted from Quarkus ExecUtil.java class at https://github.com/quarkusio/quarkus/blob/main/core/deployment/src/main/java/io/quarkus/deployment/util/ExecUtil.java
+
+package org.wildfly.plugin.provision;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.maven.plugin.logging.Log;
+
+public class ExecUtil {
+
+    private static final int PROCESS_CHECK_INTERVAL = 500;
+
+    private static class HandleOutput implements Runnable {
+
+        private final InputStream is;
+        private Log log;
+
+        HandleOutput(InputStream is, org.apache.maven.plugin.logging.Log log) {
+            this.is = is;
+            this.log = log;
+        }
+
+        @Override
+        public void run() {
+            try (InputStreamReader isr = new InputStreamReader(is);
+                 BufferedReader reader = new BufferedReader(isr)) {
+
+                for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+                    if (log != null) {
+                        log.info(line);
+                    }
+                }
+            } catch (IOException e) {
+                log.error("Failed to handle output", e);
+            }
+        }
+    }
+
+    /**
+     * Execute the specified command from within the current directory.
+     *
+     * @param command The command
+     * @param args The command arguments
+     * @return true if commands where executed successfully
+     */
+    public static boolean exec(Log log, String command, String... args) {
+        return exec(log, new File("."), command, args);
+    }
+
+    /**
+     * Execute silently the specified command until the given timeout from within the current directory.
+     *
+     * @param timeout The timeout
+     * @param command The command
+     * @param args The command arguments
+     * @return true if commands where executed successfully
+     */
+    public static boolean execSilentWithTimeout(Duration timeout, String command, String... args) {
+        return execWithTimeout(null, new File("."), timeout, command, args);
+    }
+
+    /**
+     * Execute the specified command from within the specified directory.
+     * The method allows specifying an output filter that processes the command output.
+     *
+     * @param directory The directory
+     * @param command The command
+     * @param args The command arguments
+     * @return true if commands where executed successfully
+     */
+    public static boolean exec(Log log, File directory, String command,
+                               String... args) {
+        try {
+            Process process = startProcess(directory, command, args);
+            new HandleOutput(process.getInputStream(), log).run();
```

Comment:
Is there a reason we're not launching this in a new thread? It seems like we should to ensure we get all the content.

Reply:
I definitely could be wrong, but won't it block until the process is complete? I guess this one is done without a timeout so maybe not a big deal and the next `process.waitFor()` would block anyway. It probably doesn't matter, just looked strange to me I suppose :)

Reply:
yes, this one will block until the process is complete.
There is no timeout on this one (eg builder the docker image can take as many time as it needs)